[ESQL] Async STOP / cancel / DELETE contract for EXTERNAL queries (#152585)
elastic/elasticsearch: docs/reference/query-languages/esql/esql-rest.md

This API can be used to retrieve results even if the query has already completed, as long as it's within the `keep_alive` window.
The `is_partial` field indicates result completeness. A value of `true` means the results are potentially incomplete.

> ### Stopping or cancelling queries against external sources [esql-rest-async-external-stop-cancel]

Reviewer: ```suggestion
<!--
### Stopping or cancelling queries against external sources [esql-rest-async-external-stop-cancel]
```

open comment tag

Author: Applied — opened `<!--` before the section.

When a query reads from an external source (for example, a query that begins with the `EXTERNAL` command, or any query routed to a {{esql}} data source you registered), the three ways to end the read have distinct semantics:

* **Async stop** (`POST /_query/async/{id}/stop`) signals the running query to wind down without discarding what it has already produced. Rows already accepted into the response pipeline at the moment stop arrives are returned with `is_partial: true`; rows that the source is still in the middle of delivering are dropped. If async stop races with the natural completion of a fast query, the response is returned as-is with `is_partial: false`.
* **Async delete** (`DELETE /_query/async/{id}`) cancels the underlying task before deleting the saved entry. The running query fails with a task-cancelled error; no partial body is returned.

> * **Task cancel or client disconnect** (cancellation via the [task management API](docs-content://deploy-manage/distributed-architecture/clusters-nodes-shards/task-management.md), or simply closing the HTTP connection that submitted a synchronous query) hard-fails the query with a task-cancelled error and returns no rows.

Reviewer: ```suggestion
* **Task cancel or client disconnect** (cancellation via the [task management API](esql-task-management.md), or simply closing the HTTP connection that submitted a synchronous query) hard-fails the query with a task-cancelled error and returns no rows.
```

this should fix the docs build

Author: Applied — swapped the `docs-content://` cross-repo link for the sibling `esql-task-management.md` so the docs build resolves it.

In other words, only async stop is a partial-result path for external queries — task cancel, client disconnect, and async delete all fall on the hard-fail side. This matches the behaviour for queries against {{es}} indices, with one nuance specific to external sources: a query that touches no {{es}} index has no per-cluster status to flip, so the `is_partial` flag is set directly on the response when async stop fires.

> 

Reviewer: ```suggestion
-->

```

close comment tag

Author: Applied — closed `-->` after the section. Section is now commented out end-to-end pending external-sources public rollout.
